lib/ansible/modules/storage/vexata/vexata_volume.py

```diff
@@ -0,0 +1,198 @@
+
+
+def get_volume(module, array):
+    """Retrieve a named volume if it exists, None if absent."""
+    name = module.params['name']
+    try:
+        vols = array.list_volumes()
+        vol = filter(lambda v: v['name'] == name, vols)
+        if len(vol) == 1:
+            return vol[0]
+        else:
+            return None
+    except Exception:
+        module.fail_json(msg='Error while attempting to retrieve volumes.')
+
+
+def validate_size(module, err_msg):
+    size = module.params.get('size', False)
+    if not size:
+        module.fail_json(msg=err_msg)
+    size = size_to_MiB(size)
+    if size <= 0:
+        module.fail_json(msg='Invalid volume size, must be <integer>[MGT].')
+    return size
+
+
+def create_volume(module, array):
+    """"Create a new volume."""
+    changed = False
+    size = validate_size(module, err_msg='Size is required to create volume.')
+    if module.check_mode:
+        module.exit_json(changed=changed)
+
+    try:
+        vol = array.create_volume(
+            module.params['name'],
+            'Ansible volume',
+            size)
+        if vol:
+            module.log(msg='Created volume {0}'.format(vol['id']))
+            changed = True
+        else:
+            module.fail_json(msg='Volume create failed.')
+    except Exception:
+        pass
+    module.exit_json(changed=changed)
+
+
+def update_volume(module, array, volume):
+    """Expand the volume size."""
+    changed = False
+    size = validate_size(module, err_msg='Size is required to update volume')
+    prev_size = volume['volSize']
+    if size <= prev_size:
+        module.log(msg='Volume expanded size needs to be larger '
+                       'than current size.')
+    if module.check_mode:
+        module.exit_json(changed=changed)
+
+    try:
+        vol = array.grow_volume(
+            volume['name'],
+            volume['description'],
+            volume['id'],
+            size)
+        if vol:
+            changed = True
+    except Exception:
+        pass
+
+    module.exit_json(changed=changed)
+
+
+def delete_volume(module, array, volume):
+    changed = False
+    vol_name = volume['name']
+    if module.check_mode:
+        module.exit_json(changed=changed)
+
+    try:
+        ok = array.delete_volume(
+            volume['id'])
+        if ok:
+            module.log(msg='Volume {0} deleted.'.format(vol_name))
+            changed = True
+        else:
+            raise Exception
+    except Exception:
+        pass
+    module.exit_json(changed=changed)
+
+
+def main():
+    arg_spec = argument_spec()
+    arg_spec.update(
+        dict(
+            name=dict(type='str', required=True),
+            state=dict(default='present', choices=['present', 'absent']),
+            size=dict()
+        )
+    )
+
+    module = AnsibleModule(arg_spec,
+                           supports_check_mode=True,
+                           required_together=required_together())
```

Comment:
Is anything `required_together`, if not please remove this line

Reply:
@gundalow, the username and password are required together